perfkitbenchmarker/cloudstack/cloudstack_network.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import network
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.cloudstack import util
+
+flags.DEFINE_string('cs_network_offering',
+                    'DefaultIsolatedNetworkOfferingForVpcNetworksNoLB',
+                    'Name of the network offering')
+
+flags.DEFINE_string('cs_vpc_offering',
+                    'Default VPC offering',
+                    'Name of the VPC offering')
+
+flags.DEFINE_boolean('cs_use_vpc', True,
+                     'Use VPC to create networks')
+
+
+FLAGS = flags.FLAGS
+
+
+class CloudStackFirewall(network.BaseFirewall):
+  """An object representing the Cloudstack Firewall.
+  We use static NAT hence do not need explicit
+  firewall rules all ports are open by default
+  """
+
+
+class CloudStackNetwork(network.BaseNetwork):
+  """Object representing a CloudStack Network."""
+
+  def __init__(self, zone_name):
+
+    super(CloudStackNetwork, self).__init__()
+
+    self.cs = util.CsClient(FLAGS.CS_API_URL,
+                            FLAGS.CS_API_KEY,
+                            FLAGS.CS_API_SECRET)
+
+    self.project_id = None
+
+
+    if FLAGS.project:
+        project = self.cs.get_project(FLAGS.project)
+        if project:
+            self.project_id = project['id']
+
+    self.zone_id = None
+
+    zone = self.cs.get_zone(zone_name)
+    if zone:
+        self.zone_id = zone['id']
+        self.zone = zone_name
```

Comment:
Instead of super(CloudStackNetwork, self).__init__() consider super(CloudStackNetwork, self).__init__(zone_name) so self.zone gets set to zone_name for you.